isis/tests/FunctionalTestsPointreg.cpp

```diff
@@ -0,0 +1,377 @@
+    cube2->write(line);
+    cube3->write(line);
+  }
+  cube1->close();
+  cube2->close();
+  cube3->close();
+
+  QTemporaryDir prefix;
+  QString flatFilePath = prefix.path() + "/flatfile.csv";
+  QString outNetPath = prefix.path() + "/outNet.net";
+  QVector<QString> argsA = { "fromlist=" + cubeListFile,
+                            "cnet=data/threeImageNetwork/controlnetwork.net",
+                            "deffile=data/threeImageNetwork/autoRegTemplate.def",
+                            "flatfile=" + flatFilePath,
+                            "onet=" + outNetPath,
+                            "outputfailed=no",
+                            "points=ignored" };
+  QVector<QString> argsB = { "fromlist=" + cubeListFile,
+                            "cnet=data/threeImageNetwork/controlnetwork.net",
+                            "deffile=data/threeImageNetwork/autoRegTemplate.def",
+                            "flatfile=" + flatFilePath,
+                            "onet=" + outNetPath,
+                            "measures=candidates",
+                            "points=ignored" };
+  QVector<QString> argsC = { "fromlist=" + cubeListFile,
+                            "cnet=data/threeImageNetwork/controlnetwork.net",
+                            "deffile=data/threeImageNetwork/autoRegTemplate.def",
+                            "flatfile=" + flatFilePath,
+                            "onet=" + outNetPath,
+                            "outputfailed=no",
+                            "points=nonignored" };
+  QVector<QString> argsD = { "fromlist=" + cubeListFile,
+                            "cnet=data/threeImageNetwork/controlnetwork.net",
+                            "deffile=data/threeImageNetwork/autoRegTemplate.def",
+                            "flatfile=" + flatFilePath,
+                            "onet=" + outNetPath,
+                            "outputignored=no",
+                            "points=nonignored" };
+  QVector<QString> argsE = { "fromlist=" + cubeListFile,
+                            "cnet=data/threeImageNetwork/controlnetwork.net",
+                            "deffile=data/threeImageNetwork/autoRegTemplate.def",
+                            "flatfile=" + flatFilePath,
+                            "onet=" + outNetPath,
+                            "outputignored=no",
+                            "points=ignored" };
+  UserInterface optionsA(APP_XML, argsA);  // Output ignored, register ignored
+  UserInterface optionsB(APP_XML, argsB);  // Output ignored, unmeasured, register ignored
+  UserInterface optionsC(APP_XML, argsC);  // Output ignored only
+  UserInterface optionsD(APP_XML, argsD);  // Output unmeasured only
+  UserInterface optionsE(APP_XML, argsE);  // Output unmeasured, register ignored
+  Pvl logA, logB, logC, logD, logE;
+
+  try {
+    pointreg(optionsA, &logA);
+    pointreg(optionsB, &logB);
+    pointreg(optionsC, &logC);
+    pointreg(optionsD, &logD);
+    pointreg(optionsE, &logE);
+  }
```

Comment:
usually when a Makefile has a few tests in it, they are separate tests that they felt needed to be categorized. But in the gtests, we just separate them to individual tests. I personally advocate splitting these up into four tests with exception to potentially redundant tests.

Reply:
Yeah I agree, best to stick with our convention and this way of testing these cases is pretty ugly anyways.